buildSrc/build.gradle.kts

```diff
@@ -1,7 +1,40 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+  `java-library`
+  `kotlin-dsl`
+  alias(libs.plugins.jetbrains.kotlin)
+}
+
 repositories {
+  mavenLocal()
   mavenCentral()
+  gradlePluginPortal()
 }
 
 dependencies {
+  implementation(gradleApi())
+  implementation(libs.aspectj.postCompileWeaving)
   implementation(libs.jsoup)
+  implementation(libs.jetbrains.intellij)
+  implementation(libs.jetbrains.kotlin)
+  implementation(libs.spotless)
+  implementation(libs.checkerFramework)
+}
+
+val javaMajorVersion = JavaVersion.VERSION_11
+
+project.tasks.withType<KotlinCompile> {
+//  TODO (#785): revert this setting
+//  kotlinOptions.allWarningsAsErrors = true
+
+  // Supress the warnings about different version of Kotlin used for compilation
```

Comment:
Again, probably not needed for `buildSrc`... at least I suppose these settings here do NOT apply to Kotlin compilation in the actual project, but just to `buildSrc`, right?